### ISSUE TITLE
ROM download posibility in the future?

### ISSUE THREAD

**Issue by meeeaCH** [Handheld-Database/handheld-database-gui#1]
After the GUI comes out of alpha, a good addition would be a download function. Maybe, by partnering up with a site that hosts the ROMs, and by including a direct link for the file in the DB.  (Not sure if partnering up is nessesery.) 
While you are on the device (TSP or other), you press the download and it downloads the game to the right directory.

**Comment by ogregorio** [comment]
It would be very interesting and fully possible, the problem is that we should not incur piracy, so either we support only abandonware, freeware and open source, or we let the user freely add a community list repository.

**Comment by meeeaCH** [comment]
Yes, that's why I started adding source for the game reviews too. Better to be safe, than sorry.

If the user adds the repository there can be problems with the downloads (links not pointing to the right file or something). I think the best path would be supporting the other three options you mantioned and having the option to add your source.
We would state that we don't support illegal ROM download, the function is to have an option to download your legally acquired ROMs by adding your own source. (You could have NAS where you keep them.)

**Comment by ogregorio** [comment]
I thought of a simple solution, the user would have to host a json in the format: 

{
  "system_key" : {
       "game_key" : "link",
       "platform_key" : "path to save"
  },
  "n64": {
       "mariokart64" : "link to download",
       "tsp": "path to save"
   }
}

And the system can match easily.

**Comment by meeeaCH** [comment]
Maybe we don't even need a json with this much of information. just a list of links.
We could make so the script that checks which device is the one it runs on (or you can set it up at the first launch). Than checks which system is the user currently in and gets the game name. With that knowlage it tries to download the files by going trought a list of links (provided by the user) and if it finds for example a matching name with .rar or .zip etc.. extension (the user could specify the extensions too). It downloads it and if needed it extracts it.  After that it checks if the extracted file has the right extension which is supported by the emulator.

I'll try to come up with more options/ideas.

**Comment by ogregorio** [comment]
It is a possibility. Let's study the possibility. There are even some lists of ROMs in archive.org. It would be interesting if we could extract from there.

**Comment by ogregorio** [comment]
Feature implemented.
